ui/src/dashboards/components/TemplateControlBar.tsx

```diff
@@ -16,44 +17,55 @@ interface Props {
   onSelectTemplate: (id: string) => void
 }
 
-const TemplateControlBar: SFC<Props> = ({
-  isOpen,
-  templates,
-  onSelectTemplate,
-  onOpenTemplateManager,
-  meRole,
-  isUsingAuth,
-}) => (
-  <div className={classnames('template-control-bar', {show: isOpen})}>
-    <div className="template-control--container">
-      <div className="template-control--controls">
-        {templates && templates.length ? (
-          templates.map(template => (
-            <TemplateControlDropdown
-              key={uuid.v4()}
-              meRole={meRole}
-              isUsingAuth={isUsingAuth}
-              template={template}
-              onSelectTemplate={onSelectTemplate}
-            />
-          ))
-        ) : (
-          <div className="template-control--empty" data-test="empty-state">
-            This dashboard does not have any <strong>Template Variables</strong>
+class TemplateControlBar extends Component<Props> {
+  public shouldComponentUpdate(nextProps) {
+    return !_.isEqual(this.props, nextProps)
```

Comment:
Isn't this redundant?

Reply:
this is out of scope for this PR, but no i don't think it's redundant. props have deeply nested template objects, and `shouldComponentUpdate` only does a shallow comparison. this gives it deep equality comparison to detect whether it truly needs to re-render, as it was previously closing on every autorefresh.